### PR TITLE
feat(docs): add use-cases for typeahead component

### PR DESCRIPTION
**Pull request by kulenkov**
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+typeahead/demos/async/async.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class DemoTypeaheadAsyncComponent {
   }
 
   changeTypeaheadLoading(e: boolean): void {
-    this.typeaheadLoading = e;
+     this.typeaheadLoading = e;
```
Comment (ludmilanesvitiy) [inline demo/src/app/components/+typeahead/demos/async/async.ts:90]: not need to touch src code at all

File: package-lock.json [hunk 2]
```diff
@@ -7873,6 +7886,7 @@
           "version": "1.0.2",
           "bundled": true,
           "dev": true,
+          "optional": true,
```
Comment (ludmilanesvitiy) [inline package-lock.json:7889]: you don't need to change package-lock for this changes. Your changes is related only to documentation. Need to revert other files

File: docs/spec/typeahead/typeahead.examples.with-delay.use-case.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+16.5: Typeahead With daleay example
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.with-delay.use-case.md:1]: daleay?

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+**Goal**: Show user Append to body functionality
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Append to body sub-menu
+3. User is able to fill the "Model:" input
+4. When user starts to type a name of a State from "states" array then order of typed symbols and spaces between them is ignored
+then a drop-down with a list of States matches is shown
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md:15]: in 1 line

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+5. When user starts to type a name of a State from "states" array then a drop-down with a list of States matches is shown
+6. Items in the drop-down are clickable. Click on any item auto-fills the "Model" with a selected State
+
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md:18]: 7. When user click on any item  in typeahead drop-down, then typeahead container auto-fills with a selected State

File: docs/spec/typeahead/typeahead.examples.async-data.use-case.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+1. User open Typeahead demo page
+2. User click on Async data sub-menu
+3. User is able to fill the "Model:" input. The input has a placeholder "Locations loaded with timeout"
+4. When user starts to type a name of a State from "statesComplex" array then a drop-down with a list of States matches is shown.
+Items quantity in the drop-down is limited with the key `typeaheadOptionsLimit`
+5. Items in the drop-down are clickable. Click on any item auto-fills the "Model" with a selected State
+
+<!--
+the example has an issue. it will be reported
+-->
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.async-data.use-case.md:20]: ?

File: docs/spec/typeahead/typeahead.examples.async-data.use-case.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+**Goal**: Show user Async data functionality
+
+Main success scenario:
+-------------------------
+1. User open Typeahead demo page
+2. User click on Async data sub-menu
+3. User is able to fill the "Model:" input. The input has a placeholder "Locations loaded with timeout"
+4. When user starts to type a name of a State from "statesComplex" array then a drop-down with a list of States matches is shown.
+Items quantity in the drop-down is limited with the key `typeaheadOptionsLimit`
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.async-data.use-case.md:15]: Items quantity in the drop-down is limited with the key `typeaheadOptionsLimit` - how?

File: docs/spec/typeahead/typeahead.examples.basic-array.use-case.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Basic Typeahead functionality
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Basic array sub-menu
+3. User is able to fill the "Model:" input
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.basic-array.use-case.md:13]: typeahead input

File: docs/spec/typeahead/typeahead.examples.basic-array.use-case.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+4. When user starts to type a name of a State from "states" array then a drop-down with a list of States matches is shown
+5. Items in the drop-down are clickable. Click on any item auto-fills the "Model" with a selected State
+
+Extensions:
+-----------
+3a. If there is any data, it could be deleted. While deleting data the drop-down with matches is shown
+4a. If there is no matches the drop-down is hidden. The "Model" is filled with inputted data
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.basic-array.use-case.md:20]: It's not an extension. it should be other step in the scenario

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Append to body functionality
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Append to body sub-menu
+3. User is able to fill the "Model:" input
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md:13]: User see typeahead input nd typeahead card with "Model:" text

**Review by ludmilanesvitiy — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 2 above)
Comment (kulenkov) [reply package-lock.json:7889]: i even didn't open it

**Pull request closed without merging**